docs/userguide/campaigns.md

```diff
@@ -1,25 +1,156 @@
+### Minimal definition
 
-The following additional aspects are available to further specify a campaign:
-* **A strategy:**: The strategy that is used during the campaign. As a default, a [`TwoPhaseStrategy`](baybe.strategies.composite.TwoPhaseStrategy) is employed. For more details on strategies, see [here](./strategy).
-* **Conducted experiments**: A pandas ``DataFrame`` containing the experimental representation of previously conducted experiments. Preferably, the entries of this ``DataFrame`` were created by a previous campaign using [the campaigns recommend function](baybe.campaign.Campaign.recommend). If no such ``DataFrame`` is provided, it is assumed that no experiments were conducted previously.
-* **Numerical tolerance**: This is a flag for forcing numerical measurements to be within a pre-defined tolerance. Note that the setting of the tolerances is controlled as a part of the respective parameter.
-* **Previously done batches and fits** In case that a campaign builds upon previously condiucted experiments, it is possible to provide the number of previously done batches and fits.
+When constructing a campaign, it is necessary to provide two objects:
+A search space (see [class](baybe.searchspace.core.SearchSpace) resp. [user guide](./searchspace))
+and an optimization objective (see [class](baybe.objective.Objective) resp. [user guide](./objective)).
 
-## How to use campaigns
+The search space describes the possible configuration of parameters that can be tested during the campaign.
 
-We have a detailed example explaining how to use the ``Campaign`` object in our examples, so please see [here](./../../examples/Basics/campaign) for all necessary information.
\ No newline at end of file
+```python
+from baybe.parameters import CategoricalParameter, NumericalDiscreteParameter
+from baybe.searchspace import SearchSpace
+
+cat_param = CategoricalParameter(name="cat_param", values=["a", "b", "c"])
+numdisc_param = NumericalDiscreteParameter(name="numdisc_param", values = tuple(range(1,11)))
+
+space = SearchSpace.from_product([cat_param, numdisc_param])
+```
+
+We refer to the [parameters](./parameters) resp. [search spaces](./searchspace) user
+guides for more details.
+
+For the optimization objective, we use a function taking values from 1 to 10 that we aim
+to maximize.
+
+```python
+from baybe.targets import NumericalTarget
+from baybe.objective import Objective
+
+target = NumericalTarget(name="quality", mode="MAX", bounds=(1,10))
+objective = Objective(mode="SINGLE", targets=[target])
+```
+
+Combining the search space and the objective now enables us to create a campaign.
+
+```python
+from baybe.campaign import Campaign
+
+campaign = Campaign(searchspace=space, objective=objective)
```

Comment:
I think it would be much better if already here you have the optional arguments such as strategy included with a comment ` # optional`. It makes so much clearer that there is more (compared to reading just text). That is why I have advocated for pseudo code as we really don't want to add more overhead to the already long non-campaign related code paragraph above

Reply:
This would mean to fix a way of including pseudo-code. In my opinion, we should rather have something minimal here, i.e., should explicitly not already add all of the optional stuff here - this is more advanced, and thus I'd prefer to put that into the examples and not in the very basic user guide.

@AdrianSosic , what is your opinion here?

Reply:
this is what was agreed in last dev meeting: add capability for pseudo code when it arises, imo thats now

i really dont agree with the philosophy that the user guide should leave out some optional aspects. The user guide should mention all aspects at least briefly, because it serves as the connector of someone who has no clue whats available to possibly detailed examples, otherwise its hard to understand what even exists. Theres also no reason to enforce orthogonality between user guide and examples

Reply:
I already hear somebody screaming "BOILERPLATE" regarding the non-enforcement of orthogonality 😁 
I fully agree that the user guide should not leave out optional aspects, and in my opinion, it does not leave out these aspects, as they are listed. The main disagreement here seems to be how to *present* these to the user: You seem to prefer it via the constructor, and having it all in a single place, while I prefer to have the constructor call as minimalistic as possible and explain optional arguments at another place (and not explicitly have them listed in the constructor).

Reply:
I listed the many advantages of mentioning optionals as code/pseudocode in the other thread
I see no advantages for the text version tbh so for me its a clear vote for pseudocode/code

Reply:
Ok, since the two conversations are now basically about the same topic let me resolve this conversation here.